pkg/validations/cluster.go

```diff
@@ -267,3 +271,47 @@ func ValidateManagementComponentsVersionSkew(ctx context.Context, k KubectlClien
 	}
 	return nil
 }
+
+// ValidateBottlerocketKC validates bottlerocket settings for Kubelet Configuration.
+func ValidateBottlerocketKC(spec *cluster.Spec) error {
+	cpKubeletConfig := spec.Cluster.Spec.ControlPlaneConfiguration.KubeletConfiguration
+	if err := validateKubeletConfiguration(cpKubeletConfig); err != nil {
+		return err
+	}
+
+	workerNodeGroupConfigs := spec.Cluster.Spec.WorkerNodeGroupConfigurations
+	for _, workerNodeGroupConfig := range workerNodeGroupConfigs {
+		wnKubeletConfig := workerNodeGroupConfig.KubeletConfiguration
+		if err := validateKubeletConfiguration(wnKubeletConfig); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateKubeletConfiguration(eksakubeconfig *unstructured.Unstructured) error {
+	if eksakubeconfig == nil {
+		return nil
+	}
+	var bottlerocketKC *v1beta1.BottlerocketKubernetesSettings
+
+	delete(eksakubeconfig.Object, "kind")
+	delete(eksakubeconfig.Object, "apiVersion")
```

Comment:
lets create a deepcopy and remove it from that instead of altering the original object